### PR TITLE
Fix unreliable timeout tests (issue #260)

### PR DESCRIPTION
**Pull request by markt-asf**
I have included a number of commits that just clean-up the code. I'm happy to move them to a separate PR if required.

The priority for me is the commit that fixes the tests. Not only does it reduce TCK runtime from ~30mins to ~20mins (for me, anyway) it fixes the two timeout tests that have been unreliable for as long as I have had access to the TCK.

### CODE REVIEW

**Comment by scottmarlow** [comment]
IMO, we should apply bug fixes after we can compile the Platform TCK again.

**Comment by markt-asf** [comment]
Ah. Didn't realise there was a broader compilation issue. I've been compiling individual TCKs (EL and WebSocket so far) and that has been working. I'd be happy to hold off on the clean-up as it is just cosmetic but given the standalone TCK compiles both before and after the bug fix and it only touches WebSocket sources I'd argue that it is safe to merge the bugfix commit from this PR. Let me know if you want me to update / resubmit so you have a PR with just the bug fix.

File: src/com/sun/ts/tests/websocket/ee/jakarta/websocket/websocketmessage/WSFullShortAndSessionServer.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
 public class WSFullShortAndSessionServer {
 
   @OnMessage
-  public String echo(Short s, Session session) {
```
Comment (jansupol) [inline src/com/sun/ts/tests/websocket/ee/jakarta/websocket/websocketmessage/WSFullShortAndSessionServer.java:33]: I know the session is not used. But the test is about the method with arguments Short and Session being used. See the class name: `WSFullShortAndSessionServer`. There is another endpoint `WSFullShortServer.java` that has ``echo(Short s)`. So now you have effectively duplicated a test and removed a test that verifies that a method with `Session` argument is invoked at all.

Comment (markt-asf) [reply src/com/sun/ts/tests/websocket/ee/jakarta/websocket/websocketmessage/WSFullShortAndSessionServer.java:33]: Thanks. I thought I caught all of those but I must have missed that one. I'll update that part of the clean-up and re-check the rest..
Note: I'm still happy to move the clean-up to a separate PR. It is the fix for the slow/unstable tests that is important.

Comment (jansupol) [reply src/com/sun/ts/tests/websocket/ee/jakarta/websocket/websocketmessage/WSFullShortAndSessionServer.java:33]: It looks like this is removed by mistake, all other similar cases are unchanged

Comment (jansupol) [reply src/com/sun/ts/tests/websocket/ee/jakarta/websocket/websocketmessage/WSFullShortAndSessionServer.java:33]: Yep, good work on all the tests with CountDownLatch(too_many).

**Comment by markt-asf** [comment]
I've re-based the PR (not that it really needed it but since I was going to have to force push anyway...) and fixed the issue found by Jan. I've also removed some trailing white space I had unintentionally added.

**Comment by scottmarlow** [comment]
Thanks @markt-asf for the contribution!

Thanks @jansupol for reviewing!

**Pull request merged**